grr/server/grr_response_server/gui/ui/components/client/client_test.ts

```diff
@@ -62,28 +72,60 @@ describe('Client Component', () => {
     expect(searchClientsSpy).toHaveBeenCalledWith('C.1234');
   });
 
-  it('displays client details on client change', () => {
-    // Prevent warnings from 404-ing API requests.
-    spyOn(facade, 'selectClient');
-
-    const subject = new Subject<Client>();
-    Object.defineProperty(facade, 'selectedClient$', {get: () => subject});
-
-    const fixture = TestBed.createComponent(ClientComponent);
-    fixture.detectChanges();  // Ensure ngOnInit hook completes.
-
-    paramsSubject.next(new Map(Object.entries({id: 'C.1234'})));
-    subject.next(newClient({
-      clientId: 'C.1234',
-      knowledgeBase: {
-        fqdn: 'foo.unknown',
-      },
-      lastSeenAt: new Date(1571789996678),
-    }));
-    fixture.detectChanges();
-
-    const text = fixture.debugElement.nativeElement.textContent;
-    expect(text).toContain('C.1234');
-    expect(text).toContain('foo.unknown');
-  });
+  it('correctly updates URL for client details drawer usage', fakeAsync(() => {
+       // Prevent warnings from 404-ing API requests.
+       spyOn(facade, 'selectClient');
+
+       const subject = new Subject<Client>();
+       Object.defineProperty(facade, 'selectedClient$', {get: () => subject});
+       spyOn(facade, 'removeClientLabel');
+
+       router.navigate(['v2/clients/C.1234']);
+       tick();
+       const fixture = TestBed.createComponent(ClientComponent);
+       fixture.detectChanges();  // Ensure ngOnInit hook completes.
+
+       paramsSubject.next(new Map(Object.entries({id: 'C.1234'})));
+       subject.next(newClient({
+         clientId: 'C.1234',
+         labels: [{name: 'testlabel', owner: ''}],
+       }));
+       fixture.detectChanges();
+
+       // Flows page->details page
+       expect(location.path()).toEqual('/v2/clients/C.1234');
+       expect(fixture.debugElement.query(By.directive(MatDrawer))
+                  .componentInstance.opened)
+           .toEqual(false);
+       let detailsButton =
+           fixture.debugElement.query(By.css('.goto-details')).nativeElement;
+       detailsButton.dispatchEvent(new MouseEvent('click'));
+
+       expect(location.path()).toEqual('/v2/clients/C.1234/details');
+       expect(fixture.debugElement.query(By.directive(MatDrawer))
+                  .componentInstance.opened)
+           .toEqual(true);
+
+       // Details page -> flows page
+       router.navigate(['v2/clients/C.1234/details']);
+       tick();
+       expect(location.path()).toEqual('/v2/clients/C.1234/details');
+       expect(fixture.debugElement.query(By.directive(MatDrawer))
+                  .componentInstance.opened)
+           .toEqual(true);
```

Comment:
I think refactoring this to would read more naturally and avoids awkward line wrapping:
```
const drawer = fixture.debugElement.query(By.directive(MatDrawer));
expect(drawer.componentInstance.opened).toBeTrue();
```